From 9d7e36eeb9d84440b36f8efcb1411a5f030250f0 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 04 May 2024 12:10:01 +0300
Subject: [PATCH] [SW-15774] package: control board prefix name

Zephyr's boards name template is hl_<dev>-<arc>-<app>, or
<dev>-<arc>-<app>, therefore the prefix should be controlled as well,
<prefix><dev>-<arc>-<app>

Change-Id: I74679504fb3a6071ba519a7f7d1a2c0b3a80438b
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/zephyr/fs1.mk b/package/zephyr/fs1.mk
index fcf6b72..55ee0d4 100644
--- a/package/zephyr/fs1.mk
+++ b/package/zephyr/fs1.mk
@@ -10,7 +10,7 @@
 export ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX)
 export ZEPHYR_BOARD_PREFIX = $(DEVICE_NAME)
 export SPECS_EMB_PROJ_DIR = $(SPECS_EMBEDDED_ROOT)/fs/
-
+export BOARD_PREFIX = ""
 /* TODO: remove this once mgmt application is ported */
 export ZEPHYR_MGMT_APP_DIR = $(EMBEDDED_DIR)/zephyr-apps/preboot/app/preboot_d1
 /* TODO: remove this once pid application is ported */
diff --git a/package/zephyr/gaudi2.mk b/package/zephyr/gaudi2.mk
index 7012b58..e5fef16 100644
--- a/package/zephyr/gaudi2.mk
+++ b/package/zephyr/gaudi2.mk
@@ -2,4 +2,5 @@
 export ZEPHYR_CONF_DEV_EXTRA_CFLAGS =  -I$(SPECS_EMBEDDED_DIR)
 export ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -I$(SPECS_EMBEDDED_DIR)/gaudi3/asic_reg
 export SPECS_EMB_PROJ_DIR = $(SPECS_EMBEDDED_ROOT)
+export BOARD_PREFIX = "hl_"
 endif
diff --git a/package/zephyr/gaudi3.mk b/package/zephyr/gaudi3.mk
index 26a3efd..9dda68a 100644
--- a/package/zephyr/gaudi3.mk
+++ b/package/zephyr/gaudi3.mk
@@ -7,4 +7,5 @@
 export ZEPHYR_CONF_DEV_EXTRA_CFLAGS =  -I$(SPECS_H9_DIR)
 export ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -I$(SPECS_H9_DIR)/gaudi3/asic_reg
 export SPECS_EMB_PROJ_DIR = $(SPECS_EMBEDDED_ROOT)
+export BOARD_PREFIX = "hl_"
 endif
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 5ec2378..04df8fe 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -15,7 +15,7 @@
 ZEPHYR_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
 ZEPHYR_CONF_OPTS += -DCMAKE_C_FLAGS="-I$(SPECS_EMB_PROJ_DIR) -I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX) -I$(SPECS_EMB_PROJ_DIR)/common -I$(SPECS_ROOT) -I$(SPECS_EXT_ROOT) -I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg"
 ZEPHYR_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMB_PROJ_DIR)"
-ZEPHYR_BOARD_PREFIX ?= hl_$(DEVICE_NAME)
+ZEPHYR_BOARD_PREFIX ?= $(BOARD_PREFIX)$(DEVICE_NAME)
 ZEPHYR_MGMT_APP_DIR ?= $(EMBEDDED_DIR)/zephyr-apps/arcmgmt
 ZEPHYR_PID_APP_DIR ?= $(EMBEDDED_DIR)/zephyr-apps/arcpid
 
